### DISCUSSION TITLE
Allow specifying `env.YARN_ENABLE_CONSTRAINTS_CHECKS` in the config file

### DISCUSSION THREAD

**Discussion by latin-1** [renovatebot/renovate#29995] [Suggest an Idea]
### Tell us more.

> originally posted at https://github.com/renovatebot/renovate/discussions/29287

Yarn will run `yarn constraints` on every install if `enableConstraintsChecks` is enabled. Installed packages such as `@yarnpkg/types` can be used in the constraints file (`yarn.config.cjs`). This will break renovate.

I propose allowing `env.YARN_ENABLE_CONSTRAINTS_CHECKS` in the config file and exposing it to `yarn` so that we can disable the constraints only for Renovate.

```json
{
  "env": {
    "YARN_ENABLE_CONSTRAINTS_CHECKS": "0"
  }
}
```

cc @rarkins

**Comment by rarkins** [comment]
This needs to be made generic before it can be converted to an issue

1. There's no need to restrict it to this variable only
2. Safe/allowed vars need to be admin-controlled
3. Need to define how it interacts with the existing env configuration

**Reply by latin-1** [reply]
I'm sorry, but I don't quite understand what you mean. I assumed we'd need to expand the current `allowedEnv` and `customEnvVariables` configs for the mend.io instance, but that doesn't appear to be the case, right?

**Reply by rarkins** [reply]
`customEnvVariables` is an admin/global config option, not a user/repo config option

**Reply by latin-1** [reply]
I know. What if we simply add this variable to the allow list? I don't understand what the "Generic" way is.

**Reply by rarkins** [reply]
The user equivalent is `env`: https://docs.renovatebot.com/configuration-options/#env

**Reply by latin-1** [reply]
~~I know. There might be something ambiguous. This feature request refers to the renovate GitHub app. I think your proposal primarily focuses on the self-hosted one, right?~~
Nevermind, I see the issue.

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/30015

**Reply by latin-1** [reply]
Sorry I misinterpreted your intentions. And thank you for creating this outstanding issue!

**Comment by dobesv** [comment]
We're running into this issue now - is there currently any workaround for this problem available?  Does the Mend hosted version of renovate allow setting this env var?
